tools/atlas/atlas.md

```diff
@@ -7,6 +7,55 @@ Atlas is compatible with Nimble in the sense that it supports the Nimble
 file format.
 
 
+## Concepts
+
+Atlas uses three concepts:
+
+1. Workspaces
+2. Projects
+3. Dependencies
+
+### Workspaces
+
+Every workspace is isolated, nothing is shared between workspaces.
+A workspace is a directory that has a file `atlas.workspace` inside it. If `atlas`
+is run on a (sub-)directory that is not within a workspace, a workspace is created
+automatically for you. Atlas picks the current directory or one of its parent directories
+that has no `.git` subdirectory inside it as its workspace.
+
+Thanks to this setup, it's easy to develop multiple projects at the same time.
+
+A project plus its dependencies are stored in a workspace:
+
+  $workspace / main project
+  $workspace / _deps / dependency A
+  $workspace / _deps / dependency B
+
+The deps directory can be set via `--deps:DIR` explicitly. It defaults to `_deps`.
+If you want it to be the same as the workspace use `--deps:.`.
+
+
+### Projects
+
+A workspace contains one or multiple "projects". These projects can use each other and it
+is easy to develop multiple projects at the same time.
+
+### Dependencies
+
+Inside a workspace there can be a `_deps` directory where your dependencies are kept. It is
+easy to move a dependency one level up and out the `_deps` directory, turning it into a project.
+Likewise, you can move a project to the `_deps` directory, turning it into a dependency.
+
+The only distinction between a project and a dependency is its location. For dependency resolution
+a project always has a higher priority than a dependency.
+
```

Comment:
- So `*.workspace` is a JSON?, CFG?, CSV?, something else?.
- Can I rename permanently `_deps/` to `.deps/`?.
- Atlas wont work without `*.nimble`?.

Reply:
Sounds great. :+1:

Reply:
> Virtual envs are also coming, work in progress.

Maybe those can be like plugins?, and plugins also be Atlas packages (just a `.nim` file and recompile),
and each company/team/dev can create its own custom features via plugins while keeping the core tiny.

Reply:
Well virtual envs don't fit Atlas at all currently as Atlas doesn't build anything nor does it install built executables or anything like that. It's only job is to patch a `nim.cfg` file so that `nim` can find the dependencies.